Tools/environment_install/install-prereqs-ubuntu.sh

```diff
@@ -288,12 +314,53 @@ fi
 
 # Install all packages
 $APT_GET install $BASE_PKGS $SITL_PKGS $PX4_PKGS $ARM_LINUX_PKGS $COVERAGE_PKGS
+
+heading "Check if we are inside docker environment..."
+IS_DOCKER=false
+if [[ -f /.dockerenv ]] || grep -Eq '(lxc|docker)' /proc/1/cgroup ; then
+    IS_DOCKER=true
+fi
+echo "Done!"
+
+SHELL_LOGIN=".profile"
+if $IS_DOCKER; then
+    echo "Inside docker, we add the tools path into .bashrc directly"
+    SHELL_LOGIN=".ardupilot_env"
+    echo "# ArduPilot env file. Need to be loaded by your Shell." > ~/$SHELL_LOGIN
+fi
+
+PIP_USER_ARGUMENT="--user"
+# create a Python venv on more recent releases:
+if [ ${RELEASE_CODENAME} == 'lunar' ]; then
+    $APT_GET install python3.11-venv
+    python3 -m venv venv-ardupilot
```

Comment:
python3 -m venv "$HOME/venv-ardupilot"

Reply:
and after it works !

Reply:
> Could we place it inside the repo and add it to the ignore?

That would be bad for users using Vagrant to bring up lots of different sorts of machines.

Reply:
> and after it works !

Excellent!  I've made that change.